### PR TITLE
[GEOT-5961]: Preserve alpha channel in RasterSymbolizer's ChannelSelect

### PR DESCRIPTION
**Pull request by dromagnoli**

### CODE REVIEW

File: modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/BandMergeNode.java [hunk 1]
```diff
@@ -172,8 +177,11 @@ protected GridCoverage execute() {
 
 					// color model
 					final ComponentColorModel cm = new ComponentColorModel(
-							ColorSpace.getInstance(ColorSpace.CS_LINEAR_RGB),
-							false, false, Transparency.OPAQUE,
+							(size == 1 && hasAlpha) ?
```
Comment (aaime) [inline modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/BandMergeNode.java]: Ouch, multiple inline conditional, that hurt my eyes. Could you use some temp variables instead? :-)

File: modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/BandMergeNode.java [hunk 2]
```diff
@@ -253,4 +267,8 @@ protected GridCoverage execute() {
 
 	}
 
+    public void addAlpha(RenderedImage alpha) {
```
Comment (aaime) [inline modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/BandMergeNode.java]: Javadoc missing for public method

File: modules/library/render/src/test/java/org/geotools/renderer/lite/gridcoverage2d/TransparencyStyledTest.java [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+        System.clearProperty("org.geotools.referencing.forceXY");
+        CRS.reset("all");
+    }
+
+    @BeforeClass
+    public static void init() {
+
+        // make sure CRS ordering is correct
+        CRS.reset("all");
+        System.setProperty("org.geotools.referencing.forceXY", "true");
```
Comment (aaime) [inline modules/library/render/src/test/java/org/geotools/renderer/lite/gridcoverage2d/TransparencyStyledTest.java]: I know there is a bad habit of using this one "as string" and I'm also at fault for this, but would be nicer to use GeoTools.FORCE_LONGITUDE_FIRST_AXIS_ORDER

File: modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/BandMergeNode.java (hunk 1 above)
Comment (dromagnoli) [reply modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/BandMergeNode.java]: wiil do.

File: modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/BandMergeNode.java (hunk 2 above)
Comment (dromagnoli) [reply modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/BandMergeNode.java]: will do.

**Comment by aaime** [comment]
Thank you!

**Pull request merged**